internal/characters/navia/attack.go

```diff
@@ -0,0 +1,75 @@
+package navia
+
+import (
+	"fmt"
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/geometry"
+)
+
+var (
+	attackFrames          [][]int
+	attackHitmarks        = [][]int{{23}, {22}, {31, 39, 38}, {41}}
+	attackHitlagHaltFrame = [][]float64{{0.06}, {0.06}, {0.01, 0.01, 0.01}, {.06}}
+	attackDefHalt         = [][]bool{{true}, {true}, {false, false, false}, {true}}
+	attackHitboxes        = [][]float64{{2}, {4.3, 2}, {4.5, 3}, {4.7, 2}}
+	attackOffsets         = []float64{0.5, -1.5, 1, -1.85}
+)
+
+const normalHitNum = 4
+
+func init() {
+	attackFrames = make([][]int, normalHitNum)
+
+	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], 28)
+	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1][0], 42)
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][0], 48)
```

Comment:
following Shizuka's N3 update
```
var (
    ...
    attackEarliestCancel = []int{23, 22, 29, 41}
)
... (change all first args to InitNormalCancelSlice to attackEarliestCancel[index])
attackFrames[2] = frames.InitNormalCancelSlice(attackEarliestCancel[2], 48)
attackFrames[2][action.ActionSkill] = 30
...
```
and at the bottom CanQueueAfter will need to be:
```
CanQueueAfter: attackEarliestCancel[c.NormalCounter]
```